code/modules/mob/living/carbon/human/life.dm

```diff
@@ -327,4 +331,39 @@
 		adjustOxyLoss(5)
 		adjustBruteLoss(1)
 
+/mob/living/carbon/human/proc/handle_affinity()
+	if(check_affinity(mind, AFFINITY_EARTH))

```

Comment:
There are several ways. I'm deterred from the datum approach, but it seems like it'd be better anyway so I'll see what I can do although I won't like it.